sdk/core/azure-core/src/main/java/com/azure/core/util/Contexts.java

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util;
+
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.implementation.http.HttpPipelineCallContextHelper;
+
+import java.util.Objects;
+
+/**
+ * A utility type that can be used to add and retrieve instances commonly used in {@link Context}.
+ */
+public final class Contexts {
+
+    private static final String PROGRESS_REPORTER_CONTEXT_KEY = "com.azure.core.util.ProgressReporter";
+
+    private Context context;
+
+    private Contexts(Context context) {
+        this.context = Objects.requireNonNull(context, "'context' must not be null");
+    }
+
+    /**
+     * Creates {@link Contexts} from empty {@link Context}.
+     * @return The {@link Contexts} instance.
+     */
+    public static Contexts empty() {
+        return with(Context.NONE);
+    }
+
+    /**
+     * Creates {@link Contexts} from supplied {@link Context}.
+     * @param context Existing {@link Context}. Must not be null.
+     * @return The {@link Contexts} instance.
+     * @throws NullPointerException If {@code context} is null.
+     */
+    public static Contexts with(Context context) {
+        return new Contexts(context);
+    }
+
+    /**
+     * Creates {@link Contexts} from supplied {@link HttpPipelineCallContext}.
+     * @param context Existing {@link HttpPipelineCallContext}. Must not be null.
+     * @return The {@link Contexts} instance.
+     * @throws NullPointerException If {@code context} is null.
+     */
+    public static Contexts with(HttpPipelineCallContext context) {
```

Comment:
Is this API being used anywhere yet?

Reply:
https://github.com/kasobol-msft/azure-sdk-for-java/blob/7c65a6c06ec9845ebf394077ff83e92be0950dfe/sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/policy/RequestRetryPolicy.java#L112-L115

Other option is to make `HttpPipelineCallContext.getContext` public.

Reply:
`public HttpPipelineCallContext.getContext()` sounds good.